### DISCUSSION TITLE
turbo.fish uses MiniJinja

### DISCUSSION THREAD

**Discussion by jplatte** [mitsuhiko/minijinja#379] [Show and tell]
If you are looking for a simple real-world HTTP server application that dynamically inserts user content into HTML with MiniJinja, the [source code](https://github.com/jplatte/turbo.fish/blob/6ffe2a21ccfa3a6b522e9ac1e5a0786f764bdd31/src/main.rs) for [turbo.fish](https://turbo.fish/) might be useful to you.

In particular, [`routes.rs`](https://github.com/jplatte/turbo.fish/blob/6ffe2a21ccfa3a6b522e9ac1e5a0786f764bdd31/src/routes.rs) contains [axum](https://docs.rs/axum/latest/axum/) route handlers that use the MiniJinja environment from router state to render different templates (compiled into the program for easy deploying).

**Comment by CaliViking** [comment]
@jplatte - thanks for sharing.
Are you aware of any examples with nested routes?

**Reply by mitsuhiko** [reply]
@CaliViking can you maybe share your challenges with code? It’s hard to provide suggestions without concrete challenges.

**Reply by CaliViking** [reply]
@mitsuhiko - Thank you for the extremely quick response.
I realized that I had added "loader" incorrectly to my Cargo.toml file. I updated my question to remove the add_template_owned issue.
I will try again and see if I can get it working. I'll keep you posted.

**Reply by CaliViking** [reply]
@mitsuhiko - I got everything working. This is a great Rust crate for implementing Jinja2. I am pleasantly surprised how powerful it is.
